### ISSUE TITLE
PSA: Please redirect your energies towards Ruffle

### ISSUE THREAD

**Issue by Hexstream** [pakastin/open-source-flash#143]
[Ruffle](https://ruffle.rs/) is already an advanced Flash Player emulator with great momentum, it's not complete yet but it can already play a TON of content perfectly or near perfectly. At this time, it seems Ruffle is clearly the future of Flash, so there is not much point in begging Adobe for anything regarding Flash.

(You can also download the [official Flash Player](https://www.adobe.com/support/flashplayer/debug_downloads.html) for perfect playback as a fallback option until Ruffle becomes more feature-complete.)

**edit:** Consider merging #127 or #141.

**Comment by pakastin** [comment]
Yes, Ruffle seems to work GREAT! 👌

**Comment by Alonely0** [comment]
The point of begging Adobe to open source flash is flash games are internet heritage, they are part of internet's history. Ruffle is great, and open-sourcing flash would help it A LOT, but that probably won't happen, as flash contains third-party code that would have to be relicensed too. In the case that happened, probably adobe would do it in +5 years for ensuring users don't reinstall a version without the kill switch, they must wait until flash has disappeared from the face of the internet. And in 5 years probably Ruffle will run all flash games, so there won't be any point of doing that. The best would be that Adobe contributed directly to Ruffle, but we already know Adobe is not of that kind of enterprise.

**Comment by jeffythedragonslayer** [comment]
I think the best thing to do would be if Adobe actually fixed the security
issues with Flash..  If they won't want to, but open sourced it, I'd be
more than happy to maintain it myself.

**Comment by Hexstream** [comment]
I think it's reasonable to expect that the quality of Ruffle's codebase, once it is complete, will far exceed anything Adobe could ever hope to open-source, in every dimension (security, performance, features, etc).

Also note that Ruffle is [a pretty well-funded endeavor with multiple major sponsors](https://opencollective.com/ruffle), and this is just the beginning.

I could say much more, but I encourage everyone to look into [Ruffle](https://ruffle.rs/), currently the most promising project for the future of Flash. Did I mention [Ruffle has 8.5k stars on GitHub](https://github.com/ruffle-rs/ruffle)?

**Comment by Alonely0** [comment]
> If they won't want to, but open sourced it, I'd be
more than happy to maintain it myself.

I'm pretty sure flash has a very spaghetti code since adobe didn't fix much already-known vulnerabilities and instead eventually decided to shut it down. Probably it has +100 `WONTFIX`es in the code. As I said, the only point of open-sourcing flash is to develop OSS alternatives.

**Comment by colin-i** [comment]
I have already ported my games to be playable with [Ruffle](https://github.com/ruffle-rs/ruffle).
